### PR TITLE
kvcache io kernels and test case

### PR DESCRIPTION
**Pull request by xiezhq-hermann**
## Motivation
This is a breakdown of https://github.com/sgl-project/sglang/pull/7313, upstreaming only changes on `sgl-kernel` and test cases.

## Checklist

- [x] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [x] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

### CODE REVIEW

**Review by zhyncs — APPROVED** [review]

File: sgl-kernel/tests/test_kvcacheio.py [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+        dst_pool_kernel = torch.zeros_like(dst_pool_ref)
+        dst_pool_direct = torch.zeros_like(dst_pool_ref)
+    else:
+        src_k_pool = torch.randn(
+            num_layers, total_items_in_pool, item_size
+        ).pin_memory()
+        src_v_pool = torch.randn(
+            num_layers, total_items_in_pool, item_size
+        ).pin_memory()
+        dst_k_pool_ref = torch.zeros_like(src_k_pool).to(device)
```
Comment (Edenzzzz) [inline sgl-kernel/tests/test_kvcacheio.py:78]: This is trivial, but can use `torch.zeros_like(src_k_pool, device=device)`

**Comment by Alisehen** [comment]
I have evaluated this PR using the HiCache benchmark, and it shows clear performance improvements.

For the multi-turn benchmark, I used the Qwen2.5-32B-INT8-TP2 model with 20 clients and 10 rounds.
• When the input length is 512, the PR achieves a 1.14× speedup in E2E latency and 1.4× in TTFT.
• When the input length is 1024, it achieves a 1.9× speedup in E2E and 2.5× in TTFT.

For the shared prefix benchmark, I also used Qwen2.5-32B-INT8-TP2, with 100 shuffled requests sampled from the Loogle dataset.
•With a shared prefix length of 4K, the PR improves E2E by 1.2× and TTFT by 1.4×.
•At 8K, the speedup is 1.8× for both E2E and TTFT.
•At 16K, the speedup is 1.2× for both E2E and TTFT.

**Review by zhyncs — APPROVED** [review]
